transport.go

```diff
@@ -60,6 +63,12 @@ func SetLogger(logger *logrus.Logger) {
 	sseclient.Logger = log.New(Logger.WithField("type", "sseclient").WriterLevel(logrus.TraceLevel), "", 0)
 }
 
+// SetTLSClientConfig sets the TLS configuration being used for future outbound connections.
+// A TLS configuration instance should not be modified after being set.
+func SetTLSClientConfig(config *tls.Config) {
+	tlsClientConfig = config
+}
```

Comment:
Having this support, it would be nice to add a unit test with retryable HTTP in combination with an IRMA server with TLS being enabled. This is easier when having the unit test refactors of Redis, so i'd propose to not include that here.